libcudacxx/include/cuda/std/string_view

```diff
@@ -0,0 +1,800 @@
+                "character type of basic_string_view must be standard-layout");
+  static_assert(_CCCL_TRAIT(is_trivially_default_constructible, value_type),
+                "character type of basic_string_view must be trivially default constructible");
+  static_assert(_CCCL_TRAIT(is_trivially_copyable, value_type),
+                "character type of basic_string_view must be trivially copyable");
+  static_assert(_CCCL_TRAIT(is_same, _CharT, typename traits_type::char_type),
+                "traits_type::char_type must be the same type as CharT");
+
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr basic_string_view() noexcept
+      : __data_{}
+      , __size_{}
+  {}
+
+  _CCCL_HIDE_FROM_ABI basic_string_view(const basic_string_view&) noexcept = default;
+
+  _CCCL_HIDE_FROM_ABI basic_string_view& operator=(const basic_string_view&) noexcept = default;
+
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr basic_string_view(const _CharT* __s)
+      : __data_(__s)
+      , __size_(_Traits::length(__s))
+  {}
+
+  basic_string_view(nullptr_t) = delete;
+
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr basic_string_view(const _CharT* __s, size_type __len)
+      : __data_{__s}
+      , __size_{__len}
+  {
+    // Allocations must fit in `ptrdiff_t` for pointer arithmetic to work. If `__len` exceeds it, the input
+    // range could not have been valid. Most likely the caller underflowed some arithmetic and inadvertently
+    // passed in a negative length.
+    _CCCL_ASSERT(__len <= static_cast<size_type>(numeric_limits<difference_type>::max()),
+                 "string_view::string_view(_CharT *, size_t): length does not fit in difference_type");
+    _CCCL_ASSERT(__len == 0 || __s != nullptr, "string_view::string_view(_CharT *, size_t): received nullptr");
+  }
+
+#if 0 // todo: implement range constructors
+  template <contiguous_iterator _It, sized_sentinel_for<_It> _End>
+    requires(is_same_v<iter_value_t<_It>, _CharT> && !is_convertible_v<_End, size_type>)
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr basic_string_view(_It __begin, _End __end)
+      : __data_(_CUDA_VSTD::to_address(__begin))
+      , __size_(__end - __begin)
+  {
+    _CCCL_ASSERT((__end - __begin) >= 0, "string_view::string_view(iterator, sentinel) received invalid range");
+  }
+
+  template <class _Range>
+    requires(!is_same_v<remove_cvref_t<_Range>, basic_string_view> && _CUDA_VRANGES::contiguous_range<_Range>
+             && _CUDA_VRANGES::sized_range<_Range> && is_same_v<_CUDA_VRANGES::range_value_t<_Range>, _CharT>
+             && !is_convertible_v<_Range, const _CharT*>
+             && (!requires(remove_cvref_t<_Range>& __d) { __d.operator basic_string_view<_CharT, _Traits>(); }))
+  constexpr explicit _LIBCUDACXX_HIDE_FROM_ABI basic_string_view(_Range&& __r)
+      : __data_(_CUDA_VRANGES::data(__r))
+      , __size_(_CUDA_VRANGES::size(__r))
+  {}
+#endif
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_iterator begin() const noexcept
+  {
+    return cbegin();
```

Comment:
I believe we should just do
```suggestion
    return const_iterator{__data_};
```

Applies throughout. I would also prefer if we could use direct-list-initialization when possible